src/main.rs

```diff
@@ -0,0 +1,32 @@
+mod build;
+
+use std::{
+    env,
+    path::{Path, PathBuf},
+};
+
+use anyhow::Result;
+use xshell::pushd;
+
+fn main() -> Result<()> {
```

Comment:
I would rename the file to src/xtask.rs.
Then you need to add a [bin] target to cargo.toml